megalista_dataflow/data_sources/file/file_data_source.py

```diff
@@ -0,0 +1,185 @@
+        # Get Data Source
+        data_source = self._get_data_source(execution.source.source_metadata[0])
+        # Get Data Frame
+        df = data_source.get_data_frame(execution.source.source_metadata[1])
+        # Get Uploaded Data Frame
+        df_uploaded = data_source.get_data_frame(execution.source.source_metadata[1], is_uploaded=True)
+        
+        if df is not None:
+            # Get items that haven't been processed yet
+            df_merged = df.merge(df_uploaded, how='outer')
+            df_distinct = df_merged.drop(df_merged[df_merged.timestamp.notnull()].index)
+            # Process Data Frame
+            for index, row in df_distinct.iterrows():
+                yield execution, FileDataSource._convert_row_to_dict(row)
+        else:
+            raise Exception("Unable to read from data source.")
+
+    def write_transactional_info(self, rows, execution: Execution):
+        # Get Data Source
+        data_source = self._get_data_source(execution.source.source_metadata[0])
+        # Get Data Frame
+        df = data_source.get_data_frame(execution.source.source_metadata[1], is_uploaded=True)
+        
+        now = datetime.now()
+
+        # Insert data
+        new_df = None
+        if self._transactional_type == TransactionalType.UUID:
+            new_df = pd.DataFrame([{'uuid': row['uuid'], 'timestamp': now} for row in rows])
+        elif self._transactional_type == TransactionalType.GCLID_TIME:
+            new_df = pd.DataFrame({'gclid': row['gclid'], 'time': row['time'], 'timestamp': now} for row in rows)
+        df = df.append(new_df, ignore_index=True)
+        # Upload file
+        # Add _uploaded into path
+        path = FileDataSource._append_filename_uploaded(execution.source.source_metadata[1])
+        bytes = data_source._get_file_from_data_frame(df).getbuffer().tobytes()
+        FileProvider(path, self._dataflow_options).write(bytes)
+
+    def get_data_frame(self, path: str, is_uploaded: bool = False) -> pd.DataFrame:
+        # Change filename if uploaded
+        if is_uploaded:
+            # Add _uploaded into path
+            path = FileDataSource._append_filename_uploaded(path)
+        
+        # Retrieve file
+        file = io.BytesIO(FileProvider(path, self._dataflow_options).read())
+
+        # Convert file into Data Frame
+        if file.getbuffer().nbytes == 0:
+            if is_uploaded:
+                if self._transactional_type == TransactionalType.UUID:
+                    return pd.DataFrame({'uuid': [], 'timestamp': []})
+                elif self._transactional_type == TransactionalType.GCLID_TIME:
+                    return pd.DataFrame({'gclid': [], 'time': [], 'timestamp': []})
+                else:
+                    raise NotImplementedError("Transactional type not defined.")
+            else:
+                raise ValueError(f'Unable to find data source. path="{path}"')
```

Comment:
I think we should include the Source Name  (key) here as well to indicate what source wasn't found.